src/project.cc

```diff
@@ -453,13 +454,101 @@ std::vector<std::string> ReadCompilerArgumentsFromFile(
   return args;
 }
 
+struct CCppProperties {
+  std::string cstd;
+  std::string cppstd;
+  std::vector<std::string> args;
+  bool good = false;
```

Comment:
Instead of `good` / `Load` make `Load` a non-member function that returns an optional `CCppProperties` object. That way, we remove an entire class of errors where there is an invalid `CCppProperties` instance.